src/BitBucketOnPrem.ts

```diff
@@ -25,57 +36,114 @@ export interface BitBucketOnPremInput {
     args: BitbucketOnPrem;
 }
 
+export interface LargeRepoIndicatorInput {
+    args: {
+        repoName: string,
+        host: string
+    };
+}
+
 // fetchNoCache fetches a resource without loading/saving cache and also avoids using cookies.
 // Otherwise we get inconsistent results from bitbucket API with different tokens
 const fetchNoCache = (requestInfo: RequestInfo, requestInit: RequestInit) => {
     requestInit = requestInit || {};
     if (!requestInit?.cache) {
-        requestInit.cache = 'no-store'
+        requestInit.cache = 'no-store';
     }
     if (!requestInit?.credentials) {
         requestInit.credentials = 'omit';
     }
-    return fetch(requestInfo.toString().replace('scm/', '').replace('scm%2F',''), requestInit);
-}
+    return fetch(requestInfo.toString().replace('scm/', '').replace('scm%2F', ''), requestInit);
+};
+
+export const getFileTreeByPath =
+    async ({url, accessToken, projectKey, repoName, commit, filePath}: BitbucketOnPrem): Promise<string[]> => {
+        const templateUrl: TREE_FETCH_URL = TREE_FETCH_URL.BY_PATH;
+        // build url without filePath (Url slug needs to be "src/folder" but UrlAssembler will turn it into unicode)
+        const fileTreeUrl = UrlAssembler(url).template(templateUrl)
+            .param({
+                projectKey,
+                repoName,
+            }).toString();
+
+        const sluggedUrl = addSlugToUrl(fileTreeUrl, filePath);
 
+        logger.debug("Getting files for", {projectKey, repoName, url, commit});
+        let isLastPage = false;
+        let start = 0;
+        const files: string[] = [];
+        while (!isLastPage) {
+            const res = await fetchNoCache(`${sluggedUrl}?start=${start}&limit=${FETCH_LIMIT}`, {
```

Comment:
Should you not lower the page size?

Reply:
1. the high limit is an actual performance benefit it seems in bitbucket (@DavidRBiderman  had the pleasure of figuring that one out)  
2. IMO the client should decide if its large or not and ask explorerook to set it as large, but we can discuss this one